ttt/src/game_controller.rb

```diff
@@ -63,17 +61,21 @@ def winner
     nil
   end
 
-  def take_turns
-    players.each do |player|
-      player.take_turn(self)
-      break if over?
+  def take_turn(spot = nil)
+    turn = active_player.take_turn({game: self, spot: spot})
+    if board.valid_spot?(turn.spot)
+        board.fill_spot(turn.spot)
+        turn
+    else
```

Comment:
i dont think you need an else statement here. it would return nil either way. however, we should also never explicitly return nil. there's always a better way to explain what youre returning. in this case you want to return something to indicate that the spot is not valid. could return true for a valid spot and it got filled and false for when its not.